module/spl/spl-kmem.c

```diff
@@ -133,6 +133,40 @@ strfree(char *str)
 }
 EXPORT_SYMBOL(strfree);
 
+#ifndef __GFP_REPEAT
+#define	__GFP_REPEAT __GFP_RETRY_MAYFAIL
+#endif
+void *
+spl_kvmalloc(size_t size, gfp_t flags)
+{
+	gfp_t kmalloc_flags = flags;
+	void *ret;
+#ifdef HAVE_KVMALLOC
+	if ((flags & GFP_KERNEL) == GFP_KERNEL) {
```

Comment:
I'm not sure why this conditional is needed.  Also if we intent to keep the fallback implementation, it seems like this should be an `#if, #else, #endif` block.  Or is there a case where we need the above to fallthru?  And if so, that needs to be explained in a comment.

Reply:
kvmalloc was just introduced in recent kernels. its a implementation for kernels not supporting kvmalloc. the fallthrough is required for all allocations types other than GFP_KERNEL since calling kvmalloc without GFP_KERNEL will raise a kernel warning. to support this situation the fallback implementation is used. its basicly a troubletree variant

Reply:
Got it.  If we don't provide the new `kvmem_*` interfaces then it's probably reasonable for `vmem_alloc()` to be backed by `kvmalloc()` when available, and stick with `__vmalloc()` when it's not.  Though, if we can include a fallback implementation for older kernels so much the better.